subsys/bluetooth/host/direction.c

```diff
@@ -348,3 +504,49 @@ int bt_df_adv_cte_tx_disable(struct bt_le_ext_adv *adv)
 	__ASSERT_NO_MSG(adv);
 	return bt_df_set_adv_cte_tx_enabled(adv, false);
 }
+
+#if defined(CONFIG_BT_DF_CONNECTIONLESS_CTE_RX)
+static int
+bt_df_set_per_adv_sync_cte_rx_enable(struct bt_le_per_adv_sync *sync, bool enable,
+				     const struct bt_df_per_adv_sync_cte_rx_param *params)
+{
+	if (!BT_FEAT_LE_CONNECTIONLESS_CTE_RX(bt_dev.le.features)) {
+		return -ENOTSUP;
+	}
+
+	if (!atomic_test_bit(sync->flags, BT_PER_ADV_SYNC_SYNCED)) {
+		return -EINVAL;
+	}
+
+	if (!enable &&
+	    !atomic_test_bit(sync->flags, BT_PER_ADV_SYNC_CTE_ENABLED)) {
+		return -EALREADY;
+	}
+
+	return hci_df_set_cl_cte_rx_enable(sync, enable, params);
+}
+
+int bt_df_per_adv_sync_cte_rx_enable(struct bt_le_per_adv_sync *sync,
+				     const struct bt_df_per_adv_sync_cte_rx_param *params)
+{
+	__ASSERT_NO_MSG(sync);
+	__ASSERT_NO_MSG(params);
+	return bt_df_set_per_adv_sync_cte_rx_enable(sync, true, params);
+}
+
+int bt_df_per_adv_sync_cte_rx_disable(struct bt_le_per_adv_sync *sync)
+{
+	__ASSERT_NO_MSG(sync);
+	return bt_df_set_per_adv_sync_cte_rx_enable(sync, false, NULL);
+}
+
+void bt_df_per_adv_sync_cte_cb_register(struct bt_df_per_adv_sync_cte_report_cb *cb)
+{
+	sys_slist_append(&df_per_adv_sync_cbs, &cb->node);
+}
+
+void bt_df_per_adv_sync_cte_cb_unregister(struct bt_df_per_adv_sync_cte_report_cb *cb)
+{
+	sys_slist_find_and_remove(&df_per_adv_sync_cbs, &cb->node);
+}
```

Comment:
Since you are already asserting the pointers above, perhaps also a check for `cb` here (as you do `cb->node`).

Reply:
These functions were removed because callback were added to PS sync callbacks structure.